state/metadata/state_manager.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package metadata
+
+import (
+	"bytes"
+
+	"github.com/ava-labs/avalanchego/utils/set"
+
+	"github.com/ava-labs/hypersdk/chain"
+)
+
+const (
+	defaultHeightPrefix    byte = 0x0
+	defaultTimestampPrefix byte = 0x1
+	defaultFeePrefix       byte = 0x2
+
+	DefaultMinimumPrefix byte = 0x3
+)
+
+var _ chain.MetadataManager = &MetadataManager{}
+
+type MetadataManager struct {
+	heightPrefix    []byte
+	feePrefix       []byte
+	timestampPrefix []byte
+}
+
+func NewDefaultManager() MetadataManager {
+	return MetadataManager{
+		heightPrefix:    []byte{defaultHeightPrefix},
+		feePrefix:       []byte{defaultFeePrefix},
+		timestampPrefix: []byte{defaultTimestampPrefix},
+	}
+}
+
+func (m MetadataManager) FeePrefix() []byte {
+	return m.feePrefix
+}
+
+func (m MetadataManager) HeightPrefix() []byte {
+	return m.heightPrefix
+}
+
+func (m MetadataManager) TimestampPrefix() []byte {
+	return m.timestampPrefix
+}
+
+// Returns true if all prefixes in `m` and `vmPrefixes` are unique
+func HasConflictingPrefixes(
+	m chain.MetadataManager,
+	vmPrefixes [][]byte,
+) bool {
+	prefixes := [][]byte{
+		m.HeightPrefix(),
+		m.FeePrefix(),
+		m.TimestampPrefix(),
+	}
+
+	prefixes = append(prefixes, vmPrefixes...)
+	verifiedPrefixes := set.Set[string]{}
+
+	for _, p := range prefixes {
+		for vp := range verifiedPrefixes {
+			if bytes.HasPrefix(p, []byte(vp)) || bytes.HasPrefix([]byte(vp), p) {
+				return true
+			}
+		}
+
+		verifiedPrefixes.Add(string(p))
+	}
```

Comment:
Is there any reason to use `verifiedPrefixes` as a set here? It seems like we're just using it as a list. The only change this would add is we'd make sure not to check against the same verified prefix more than once, but if we hit any duplicates we should already exit early.

Could we add a short unit test here that covers the cases:
1. no conflicts
2. identical prefixes
3. prefixes contains a prefix of one of the vm prefixes
4. vm prefix contains a prefix of one of the prefixes
5. vm prefixes contains a duplicate

Reply:
Done

Reply:
Do we need to use a set for `verifiedPrefixes` here rather than a slice?

Reply:
I used `set.Set` here as it was originally used in the state-factory PR found here (and therefore thought it was the preferred style) https://github.com/ava-labs/hypersdk/blob/a6fdf33ea39140ca0ae3bec5cd63aa51d6fa9dfd/state/layout.go#L100-L112 

I have no preference as to the DS being used and will switch to using a slice instead